base/server/healthcheck/pki/server/healthcheck/clones/plugin.py

```diff
@@ -0,0 +1,310 @@
+            self.passwd = self.instance.get_password(self.subsystem_token)
+
+        return sec_domain, sechost, secport
+
+    def check_ca_clones(self):
+        for host in self.clone_cas:
+            cur_clone_msg = ' Host: ' + host.Hostname + ' Port: ' + host.SecurePort
+            # Reach out and get some certs, to serve as a data and connectivity check
+            try:
+                connection = PKIConnection(protocol='https',
+                                           hostname=host.Hostname,
+                                           port=host.SecurePort,
+                                           verify=False)
+
+                cert_client = CertClient(connection)
+                # get the first 5 in case we cant to make a sanity check of replicated data
+                certs = cert_client.list_certs(size=5)
+
+                if certs is not None and len(certs.cert_data_info_list) == 5:
+                    logger.info('Cert data successfully obtained from clone.')
+                else:
+                    raise BaseException('CA clone problem reading data.' + cur_clone_msg)
+            except BaseException as e:
+                logger.error("Internal server error %s", e)
+                raise BaseException('Internal error testing CA clone.' + cur_clone_msg)
+
+        return
+
+    def check_kra_clones(self):
+        for host in self.clone_kras:
+            cur_clone_msg = ' Host: ' + host.Hostname + ' Port: ' + host.SecurePort
+            # Reach out and get some keys or requests , to serve as a data and connectivity check
+            try:
+                client_nick = self.security_domain.config.get('ca.connector.KRA.nickName')
+
+                output = self.contact_subsystem_using_pki(
+                    host.SecurePort, host.Hostname, client_nick,
+                    self.passwd, self.db_dir, 'kra-key-show', ['0x01'])
+
+                # check to see if we either got a key or a key not found exception
+                # of which either will imply a successful connection
+                if output is not None:
+                    key_found = output.find('Key ID:')
+                    key_not_found = output.find('KeyNotFoundException:')
+                    if key_found >= 0:
+                        logger.info('Key material found from kra clone.')
+
+                    if key_not_found >= 0:
+                        logger.info('key not found, possibly empty kra')
+
+                    if key_not_found == -1 and key_found == -1:
+                        logger.info('Failure to get key material from kra')
+                        raise BaseException('KRA clone problem detected ' + cur_clone_msg)
+                else:
+                    raise BaseException('No data obtained from KRA clone.' + cur_clone_msg)
+
+            except BaseException as e:
+                logger.error("Internal server error %s", e)
```

Comment:
(nitpick) The term "Internal Server Error" seems to correspond to response code 500. We should use some other log message to keep the user clear. May be "Clone issue: %s, e"? WDYT?

Reply:
Yes, agreed, will do.

Reply:
Taken care of in a few places as requested.